src/cdi/CPEloss.hh

```diff
@@ -34,6 +34,8 @@ namespace rtt_cdi {
 
 class CPEloss {
 public:
+  CPEloss(CParticle target_in, CParticle projectile_in)
```

Comment:
I've been thinking about this constructor (again :D), and I have a question: in the case of tabular data, is there any reason not to just use the particle masses found in the data tables? I'm trying to figure out if this general constructor format (passing in particle objects that already "know" their own basic properties) makes sense for all the inherited Eloss classes, or if we should modify it somehow. In the case of the tabular data, it SEEMS like all we really need to know are the ZAIDs, and the rest can be gleaned from the table... but is there something I'm missing here?

Reply:
The tabular data format read by this class does provide target and projectiles masses (although not the mass of the electron target!) that we could read in. I guess I feel safer though keeping constants like particle masses centralized somewhere in the code, so the code doesn't one day find itself in a situation where a particle has a different mass depending on which line of code it's on. If this constructor has problems though I'm happy to discuss. Another solution could be to have a more global namespace-y depot that can be queried for things like particle masses (and then you would only have to pass ZAIDs to Eloss classes).

Reply:
Cool, I'm fine with your reasoning (and leaving the constructor as-is, at least until someone else complains about it :) ).